spec/index.bs

```diff
@@ -2517,6 +2518,82 @@ dictionary GPUImageBitmapCopyView {
+  - (|destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/z}} + |copySize|.{{GPUExtent3DDict/depth}}) must not overflow a {{GPUIntegerCoordinate}}.
+  - (|destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/x}} + |copySize|.{{GPUExtent3DDict/width}}) must be less than or equal to the {{GPUExtent3DDict/width}} of the [=physical size=] of
+    |destination|.{{GPUTextureCopyView/texture}} [=subresource=] at [=mipmap level=] {{GPUTextureCopyView/mipLevel}} and the [=array layer=] {{GPUTexture/[[textureSize]]}}.{{GPUExtent3DDict/depth}}
+    when the {{GPUTexture/[[dimension]]}} of {{GPUTextureCopyView/texture}} is {{GPUTextureDimension/2d}}.
+  - (|destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/y}} + |copySize|.{{GPUExtent3DDict/height}}) must be less than or equal to the {{GPUExtent3DDict/width}} of the [=physical size=] of
+    |destination|.{{GPUTextureCopyView/texture}} [=subresource=] at [=mipmap level=] {{GPUTextureCopyView/mipLevel}} and the [=array layer=] {{GPUTexture/[[textureSize]]}}.{{GPUExtent3DDict/depth}}
+    when the {{GPUTexture/[[dimension]]}} of {{GPUTextureCopyView/texture}} is {{GPUTextureDimension/2d}}.
+  - |source|.{{GPUBufferCopyView/buffer}}.{{GPUBuffer/[[size]]}} must be greater than or equal to |source|.{{GPUBufferCopyView/offset}}.
+  - (|source|.{{GPUBufferCopyView/buffer}}.{{GPUBuffer/[[size]]}} - |source|.{{GPUBufferCopyView/offset}}) must be greater than or equal to the number of |required bytes in copy|.
+  - |source|.{{GPUBufferCopyView/offset}} must be a multiple of the [=texel block size=] of the {{GPUTexture/[[format]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+
+  The |bytes in a complete row| is calculated with the following algorithm. An error occurs when there is any overflow in the calculation of |bytes in a complete row|:
+  ```
+  bytesInACompleteRow = copySize.width / texelBlockWidth * texelBlockSize;
+  ```
+
+  The |required bytes in copy| is calculated with the following algorithm. An error occurs when there is any overflow in the calculation of |required bytes in copy|:
+  ```
+  if (copySize.height == 0 || copySize.depth == 0) {
+      bytesRequiredInTheCopy = 0;
+  } else {
+      GPUSize64 trueBytesPerRow = (bytesPerRow == 0) ? (copySize.width / texelBlockWidth * texelBlockSize) : bytesPerRow;
+      GPUSize64 trueRowsPerImage = (rowsPerImage == 0) ? copySize.height : rowsPerImage;
+      GPUSize64 bytesPerImage = trueBytesPerRow * trueRowsPerImage;
+      GPUSize64 bytesInLastSlice = trueBytesPerRow * (copySize.height / texelBlockHeight - 1) + (copySize.width / texelBlockWidth) * texelBlockSize;
+      bytesRequiredInTheCopy = bytesPerImage * (copySize.depth - 1) + bytesInLastSlice;
+  }
+  ```
+
+Issue(gpuweb/gpuweb#69): Define the copies with {{GPUTextureDimension/1d}} and {{GPUTextureDimension/3d}} textures.
```

Comment:
nit: Also add a TODO to decide what to do for depth/stencil formats?

Reply:
Done